01-front-end-fundamentals/git-ssh/readme.md

```diff
@@ -0,0 +1,161 @@
+# Generating SSH Keys and Linking with Github
+
+## Learning Objectives
+* Explain how SSH serves as extra layer of security
+* Define and Differentiate between HTTPS and SSH
+* List the advantages of an SSH connection
+
+## Opening
+Secure Shell (SSH) is a command interface and protocol for securely getting access to a remote computer.
+SSH uses public-key cryptography to authenticate the remote computer and allow it to authenticate the user, if necessary. Both ends of the client/server connection are authenticated using a digital certificate, and passwords are protected by being encrypted.
+
+## Step 0: Checking for existing Keys
+First, we need to check for existing SSH keys on your computer. Open Terminal and enter:
+
+  ```
+  $ ls -al ~/.ssh
+  # Lists the files in your .ssh directory, if they exist
+  ```
+
+If you already have a key, skip to step 2. Otherwise proceed to step 1.
+
+
+## Step 1: Generate a new SSH Key
+1. With Terminal still open, copy and paste the text below. Make sure you substitute in your GitHub email address.
+
+  ```
+  $ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+  # Creates a new ssh key, using the provided email as a label
+  # Generating public/private rsa key pair.
+  ```
+
+2. We strongly suggest keeping the default settings as they are, so when you're prompted to "Enter a file in which to save the key", just press Enter to continue.
+
+  ```
+  Enter file in which to save the key (/Users/you/.ssh/id_rsa): [Press enter]
+  ```
+
+3. You'll be asked to enter a passphrase.
+
+  ```
+  Enter passphrase (empty for no passphrase): [Type a passphrase]
+  # Enter same passphrase again: [Type passphrase again]
+  ```
+
+4. After you enter a passphrase, you'll be given the fingerprint, or id, of your SSH key. It will look something like this:
+
+  ```
+  Your identification has been saved in /Users/you/.ssh/id_rsa.
+  # Your public key has been saved in /Users/you/.ssh/id_rsa.pub.
+  # The key fingerprint is:
+  # 01:0f:f4:3b:ca:85:d6:17:a1:7d:f0:68:9d:f0:a2:db your_email@example.com
+  ```
+Great now that we have created a new key, we need to add it to the ssh-agent so we can easily access it.
+
+## Step 3: Add Your Key to the SSH-Agent
+To configure the ssh-agent program to use your SSH key:
+1. Ensure ssh-agent is enabled:
+
+  ```
+  # start the ssh-agent in the background
+  $ eval "$(ssh-agent -s)"
+  # Agent pid 59566
+  ```
```

Comment:
I do not think it is necessary-just an extra step to manage your keys in the future, recommended by Github. Open to cut it out if you think it make its more confusing.